File: DogShelter/DogShelter.Model/Favorit.cs
namespace DogShelter.Model;

public class Favorit
{
    public int FavoritId { get; set; }
    public int KorisnikId { get; set; }
    public int PasId { get; set; }
    public DateTime DatumDodavanja { get; set; }

    // Denormalized dog display fields so "Moji favoriti" can render a card straight off this
    // list, the same way PreporuceniPas carries what it needs instead of a second round-trip.
    public string PasNaziv { get; set; } = null!;
    public string? RasaNaziv { get; set; }
    public string? VelicinaNaziv { get; set; }
    public string SlikaNaslovna { get; set; } = null!;
    public string? StatusNaziv { get; set; }
    public bool PasAktivan { get; set; }
}

File: dogshelter_shared/lib/favorit/domain/favorit.dart
class Favorit {
  Favorit({
    required this.favoritId,
    required this.korisnikId,
    required this.pasId,
    required this.datumDodavanja,
    required this.pasNaziv,
    this.rasaNaziv,
    this.velicinaNaziv,
    required this.slikaNaslovna,
    this.statusNaziv,
    required this.pasAktivan,
  });

  final int favoritId;
  final int korisnikId;
  final int pasId;
  final DateTime datumDodavanja;
  final String pasNaziv;
  final String? rasaNaziv;
  final String? velicinaNaziv;
  final String slikaNaslovna;
  final String? statusNaziv;
  final bool pasAktivan;

  factory Favorit.fromJson(Map<String, dynamic> json) {
    return Favorit(
      favoritId: json['favoritId'] as int,
      korisnikId: json['korisnikId'] as int,
      pasId: json['pasId'] as int,
      datumDodavanja: DateTime.parse(json['datumDodavanja'] as String),
      pasNaziv: json['pasNaziv'] as String,
      rasaNaziv: json['rasaNaziv'] as String?,
      velicinaNaziv: json['velicinaNaziv'] as String?,
      slikaNaslovna: json['slikaNaslovna'] as String,
      statusNaziv: json['statusNaziv'] as String?,
      pasAktivan: json['pasAktivan'] as bool,
    );
  }
}
